app/utils/repository-poller.ts

```diff
@@ -18,6 +18,7 @@ export default class RepositoryPoller extends Poller {
     'user.badge-awards.course-stage-completion',
     'user.badge-awards.course-stage-completion.course-stage',
     'user.badge-awards.course-stage-completion.repository',
+    'user.promotional-discounts',
```

Comment:
@ryan-gang we shouldn't need this, `syncUser` should just work. Repository#index and User#index are called very very often, so we should minimize what they request as much as possible